utils/box-symbol.stanza

```diff
@@ -244,6 +244,22 @@ public defn make-box-symbol (banks:Seqable<BoxBank>) :
 ;=======================================================<doc>
 
 public defn make-box-symbol () :
+  ;Sort the given pins by their .pin-properties-row-index field.
+  defn sort-by-row-index (pins:Tuple<JITXObject>) -> Tuple<JITXObject> :
+    ;Compare pin[i0] versus pin[i1].
+    ;Indices are used, instead of the pin directly, in order to perform a
+    ;stable sort.
+    ;All pins without a .pin-properties-row-index is put in the beginning.
+    defn compare-index (i0:Int, i1:Int) :
+      val a = pins[i0]
+      val b = pins[i1]
+      match(value?(property?(a.pin-properties-row-index)), value?(property?(b.pin-properties-row-index))) :
+        (ai:Int, bi:Int) : compare(ai, bi)
+        (ai:Int, bi) : 1
+        (ai, bi:Int) : -1
+        (ai, bi) : compare(i0,i1)
+    val indices = qsort(0 to length(pins), compare-index)
```

Comment:
`qsort` or `qsort!`

Reply:
`qsort!` modifies the passed-in collection.
`qsort` returns a new collection.